typhos/notes.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import os
+from datetime import datetime
+from enum import IntEnum
+from pathlib import Path
+from typing import Dict, Tuple
+
+import platformdirs
+import yaml
+from qtpy import QtWidgets
+
+from typhos import utils
+
+logger = logging.getLogger(__name__)
+
+
+class NotesSource(IntEnum):
+    USER = 0
+    ENV = 1
+    HAPPI = 2
+
+
+def get_notes_data(device_name: str) -> Tuple[NotesSource, Dict[str, str]]:
+    """
+    get the notes data for a given device
+    attempt to get the info from the following locations
+    in order of priority (higher priority will shadow)
+    1: device_notes in the user directory
+    2: the path in DEVICE_NOTES environment variable
+    3: TODO: happi ... eventually
+
+    Parameters
+    ----------
+    device_name : str
+        The name of the device to retrieve notes for.
+
+    Returns
+    -------
+    Tuple[NotesSource, dict]
+        The source of the device notes
+        a dictionary containing the device note information
+    """
+    data = {'note': '', 'timestamp': ''}
+    source = NotesSource.USER
+    # try user directory
+    user_data_path = platformdirs.user_data_path() / 'device_notes.yaml'
+    if user_data_path.exists():
+        with open(user_data_path) as f:
+            device_notes = yaml.full_load(f)
+
+        if device_name in device_notes:
+            source = NotesSource.USER
+            data = device_notes.get(device_name)
+
+    env_notes_path = os.environ.get('DEVICE_NOTES')
```

Comment:
Can we tie this filename to typhos a bit more?
Can we share this information with lucid and use the same file?

Since lucid depends on typhos, it'd be OK if lucid supports typhos's format (and not the other way around) I think.

Reply:
Perhaps in the interest of using this for multiple apps we keep this more general?  (since we talked about using this apps like hutch-python, etc)

Reply:
An environment variable without a shared prefix still doesn't sound great to me (even if we don't make it `TYPHOS_` something). It leaves me not knowing what to grep for, or what goes along with what when I type `env` to look at everything we define in our environments (which is a lot).

I know this sounds nitpicky on the surface, but changing it going forward can be also be painful/confusing. 
... So, can we do better than `DEVICE_NOTES`?

Reply:
Fair point.  But names are hard.  We don't currently have a prefix for variables shared between our libraries, which is what I think we want here.  Thoughts on:
* `PCDS_DEVICE_NOTES`: risks making alex sad about dead-naming our group, but matches our pcdshub org
* `ECS_DEVICE_NOTES`: more "modern", but it feels like I'm claiming too much... maybe that's just a me problem
* `LCLS_DEVICE_NOTES`: definitely too general of an environment variable

Reply:
My initial thought is that environment variables are easy enough to tweak that something like `TYPHOS_DEVICE_NOTES` would be totally OK. If this migrates to happi and becomes `HAPPI_DEVICE_NOTES` or if it goes away entirely and gets consumed by happi it'll be ok, and if we need to push it up to something like `pcdsutils` it can be renamed easily too. We can even continue to support `TYPHOS_DEVICE_NOTES` as a backup in whatever capacity we move it to.

Note that typhos doesn't necessarily imply pcds/ecs as well.

Reply:
As long as we decide on something before people start setting env vars in their various .bashrc's and don't have to go through correcting them, I think I agree.  Changing things in our control is easy enough, it's the general public I'm more worried about

Reply:
I have chosen `PCDS_DEVICE_NOTES`.  Hopefully future me will look back on this moment and not be too annoyed